414 merge am3 miscellaneous science modifications

JhanSrbinovsky opened the pull request:
# CABLE

Thank you for submitting a pull request to the CABLE Project.

## Description

Miscellaneous developments in science/ dir. from AM3. These are included separately to more major developments in canopy and soilsnow directories.

Fixes #402

## Type of change

Including AM3 development.

## Checklist

- [X ] The new content is accessible and located in the appropriate section.
- [ X] I have checked that links are valid and point to the intended content.
- [ X] I have checked my code/text and corrected any misspellings

<!-- readthedocs-preview cable start -->
----
📚 Documentation preview 📚: https://cable--430.org.readthedocs.build/en/430/

<!-- readthedocs-preview cable end -->

JhanSrbinovsky commented:
Not ready to submit for merge as dependent on preceding merge of #402, params

see [benchcab analysis](https://modelevaluation.org/analyses/workspace/)

JhanSrbinovsky commented:
Note: comparison here is to branch #402

har917 commented on src/science/radiation/cbl_init_radiation.F90:
```diff
@@ -378,9 +378,8 @@ SUBROUTINE BeamFraction( RadFbeam, mp, nrb, Cpi,Ccoszen_tols_huge, metDoy,     &
-RadFbeam(:,1) = spitter(mp, cpi, metDoy, coszen, SW_down(:,1)+SW_down(:,2))
-RadFbeam(:,2) = RadFbeam(:,1)
+RadFbeam(:,1) = spitter(mp, cpi, metDoy, coszen, SW_down(:,1))
+RadfBeam(:,2) = spitter(mp, cpi, metDoy, coszen, SW_down(:,2))
```
Please check with @rml599gh - I think this is reinstating a bug that we identified in CABLE3-ESM1.6 development and so the change needs to go the other way.

rml599gh replied:
Yes, #355, #354 describe this bug. Code in red is the correct version. @har917 - great that you picked this up. Highlights the value of reviews.

JhanSrbinovsky replied:
fixed, sorry - IDK how this happened

har917 commented on src/science/albedo/cbl_snow_albedo.F90:
```diff
@@ -76,6 +76,9 @@ SUBROUTINE surface_albedosn( AlbSnow, AlbSoil, mp, nrb, ICE_SoilType,          &
      alvo  = 0.95,  &  ! albedo for vis. on a new snow
      aliro = 0.70      ! albedo for near-infr. on a new snow
 
+REAL, PARAMETER :: sfact_default = 0.68
```
changes relate to code standards - no change in science - good to go in.

har917 commented on src/science/misc/cable_carbon.F90:
```diff
@@ -150,8 +155,16 @@ SUBROUTINE carbon_pl(dels, soil, ssnow, veg, canopy, bgc)
     ! drought stress
-    coef_drght = EXP( 5.*( MIN( 1., MAX( 1., wbav**( 2 - soil%ibp2 ) - 1.) /    &
-         ( soil%swilt**( 2 - soil%ibp2 ) - 1. ) ) - 1. ) )
+    CampbellExp         = 2.0 - soil%ibp2
```
I wouldn't use `CampbellExp` as the working variable name - unless it is exactly the same (and same science intent) as the Campbell parts of the soil moisture science.

Otherwise the break up of the equation is good (possibly over the top)

har917 commented on src/science/roughness/cable_roughness.F90:
```diff
@@ -22,17 +23,22 @@
-USE cable_phys_constants_mod, ONLY : CZDLIN  => ZDLIN
-USE cable_phys_constants_mod, ONLY : CCSW    => CSW
-USE cable_phys_constants_mod, ONLY : CGRAV   => GRAV
+#ifdef UM_CBL
```
`#ifdef` okay for now - but surely (for this one) we can get round the need for this by modifying the files names in the offline/coupled codebases to be the same.

har917 commented on src/science/roughness/roughnessHGT_effLAI_cbl.F90:
```diff
@@ -32,6 +32,12 @@ MODULE hruff_eff_LAI_mod_cbl
+!* The first procedure in this module evaluates the canopy height
+!  given the effect of any snow present.
+
+!* The secone procedure in this module computes the effective LAI of a canopy
```
typo - 'secone' should be 'second'

JhanSrbinovsky commented:
One last thing - did you look at the benchcab analysis? I couldn't really make any sense of it. Some slightly better, some slightly worse it seemed.

JhanSrbinovsky commented:
DOH!! Because of the lag in submitting these to main, and they are all sequential , and somewhat dependent on the previous tranche of developments, I branched this (or at least compared it to the pre-ceding, related issue #402 ). Unfortunately this hadn't gone through, I hadn't remembered weeks later that I had "compared" to the 402 branch in this PR and so when merging was approved/executed - it went back to the params branch.

In the interim I have addressed the comments about cable_surface_types in #436. This PR will/should go through fairly quickly. Regardless, I need to, it will be cleaner if I rebase from main post #436 going in AND then re-submit this PR.
